code/modules/mob/living/basic/slime/feeding.dm

```diff
@@ -0,0 +1,154 @@
+
+//Feeding
+
+///Can the slime leech life energy from the target?
+/mob/living/basic/slime/proc/can_feed_on(mob/living/meal, silent = FALSE, check_adjacenct = FALSE)
+
+	if(!isliving(meal)) //sanity check
+		return FALSE
+
+	if(stat)
+		if(silent)
+			return FALSE
+		balloon_alert(src, "unconscious!")
+		return FALSE
+
+	if(hunger_disabled)
+		if(silent)
+			return FALSE
+		balloon_alert(src, "not hungry!")
+		return FALSE
+
+	if(check_adjacenct && !Adjacent(meal))
+		return FALSE
+
+	if(meal.stat == DEAD)
+		if(silent)
+			return FALSE
+		balloon_alert(src, "no life energy!")
+		return FALSE
+
+	if(locate(/mob/living/basic/slime) in meal.buckled_mobs)
+		if(silent)
+			return FALSE
+		balloon_alert(src, "another slime in the way!")
+		return FALSE
+
+	if(issilicon(meal) || meal.mob_biotypes & MOB_ROBOTIC || meal.flags_1 & HOLOGRAM_1)
+		balloon_alert(src, "no life energy!")
+		return FALSE
+
+	if(isslime(meal))
+		if(silent)
+			return FALSE
+		balloon_alert(src, "can't eat slime!")
+		return FALSE
+
+	if(isanimal(meal))
+		var/mob/living/simple_animal/simple_meal = meal
+		if(simple_meal.damage_coeff[TOX] <= 0 && simple_meal.damage_coeff[BRUTE] <= 0) //The creature wouldn't take any damage, it must be too weird even for us.
+			if(silent)
+				return FALSE
+			balloon_alert(src, "not food!")
+			return FALSE
+	else if(isbasicmob(meal))
+		var/mob/living/basic/basic_meal = meal
+		if(basic_meal.damage_coeff[TOX] <= 0 && basic_meal.damage_coeff[BRUTE] <= 0)
+			if (silent)
+				return FALSE
+			balloon_alert(src, "not food!")
+			return FALSE
+
+	return TRUE
+
+///The slime consumes the mob's lifeforce
+/mob/living/basic/slime/proc/feed_process(seconds_per_tick = SSMOBS_DT)
+	if(!isliving(buckled)) //Just in case
+		stop_feeding(silent = TRUE)
+
+	var/mob/living/prey = buckled
+
+	if(stat)
+		stop_feeding(silent = TRUE)
+
+	if(prey.stat == DEAD) // our victim died
+		if(client)
+			to_chat(src, span_info("This subject does not have a strong enough life energy anymore..."))
+
+		if(ai_controller && !ai_controller.blackboard[BB_SLIME_RABID])
+			var/mob/last_to_hurt = prey.LAssailant?.resolve()
```

Comment:
`LAssailant` should almost certainly be replaced by some blackboard key

Reply:
I was about to say, that is probably out of scope here, but then I have found out that is literally the only place where this is read instead of being set. 

It could be replaced by the retaliate list of the mob, but not all mobs have the retaliate element, and also its a list instead of a specific mob.

Reply:
Whoops reproduce() is used by a crossbreed. I'll leave that alone for now, but will rework it later when that crossbreed is gone.

Reply:
I have thought a long time about this, but could not figure out an universal solution :(

Reply:
Paraphrasing from last night's discussion: this will be unique to the slime management consoles, done via a status effect. Possibly will add some particles.

Reply:
It is done.